DEVELOPMENT.md

```diff
@@ -124,6 +124,19 @@ Enable errorprone ([Error Prone Installation#IDEA](https://errorprone.info/docs/
 - In ``Java Compiler`` tab, select ``Javac with error-prone`` as the compiler,
 - Update ``Additional command line parameters`` with ``-XepExcludedPaths:.*/target/generated-(|test-)sources/.* -XepDisableAllChecks -Xep:MissingOverride:ERROR ......`` (for current recommended list of command line parameters, see the top level ``pom.xml``, the definition of the ``errorprone-compiler`` profile.
 
+### Language injection in IDE
+
+In order to enable language injection inside Intellij IDEA, some code elements can be annotated with the `@org.intellij.lang.annotations.Language` annotation. To make it useful, we recommend:
+
+- Set the project-wide SQL dialect in ``Languages & Frameworks | SQL Dialects`` - "Generic SQL" is a decent choice here,
+- Disable inspection ``SQL | No data source configured``,
+- Optionally disable inspection ``Language injection | Language mismatch``.
+
+Even if the IDE does not support language injection this annotation is useful for documenting the API's intent. Considering the above, we recommend annotating with `@Language`:
+
+- All API parameters which are expecting to take a `String` containing an SQL statement (or any other language, like regular expressions),
+- Local variables which otherwise would not be properly recognized by IDE for language injection.
```

Comment:
1."which otherwise would not be properly recognized by IDE"
Just thinking logically, how a person not having InteliJ Ultimate knows what that means?

2.So we want to make annotations in API and annotation before every local variable an equally recommended thing. The current state of the code does not represent this - there is a lot of args with this annotation and almost no variables.

Reply:
>  how a person not having InteliJ Ultimate knows what that means

They don't know, so they don't care, so they don't change.

> So we want to make annotations in API and annotation before every local variable an equally recommended thing.

No. Only in cases where `Local variables which otherwise would not be properly recognized by IDE for language injection.` otherwise annotation should not be added. I expect `a lot of args with this annotation and almost no variables.`

Reply:
> They don't know, so they don't care, so they don't change.
That sounds terrible, but I guess you are right. I don't understand it and already started not caring.